test/test_utils.py

```diff
@@ -2015,6 +2015,27 @@ def test_traverse_obj(self):
                          msg='function as query key should perform a filter based on (key, value)')
         self.assertCountEqual(traverse_obj(_TEST_DATA, lambda _, x: isinstance(x[0], str)), {'str'},
                               msg='exceptions in the query function should be catched')
+        if __debug__:
```

Comment:
```suggestion

        # Test wrong function signature
        if __debug__:
```

Reply:
How about embedding the comment in the assertion message?
```py
        if __debug__:
            with self.assertRaises(Exception, msg='Wrong function signature should raise in debug'):
                traverse_obj(_TEST_DATA, lambda a: ...)
            with self.assertRaises(Exception, msg='Wrong function signature should raise in debug'):
                traverse_obj(_TEST_DATA, lambda a, b, c: ...)
```

Reply:
I suggested this coz you have comments for the other sections

Reply:
But this is the same section (Query functions)?